src/PollingBlockTracker.ts

```diff
@@ -303,25 +279,53 @@ export class PollingBlockTracker
     }
     return res.result;
   }
-}
 
-/**
- * Waits for the specified amount of time.
- *
- * @param duration - The amount of time in milliseconds.
- * @param unref - Assuming this function is run in a Node context, governs
- * whether Node should wait before the `setTimeout` has completed before ending
- * the process (true for no, false for yes). Defaults to false.
- * @returns A promise that can be used to wait.
- */
-async function timeout(duration: number, unref: boolean) {
-  return new Promise((resolve) => {
-    const timeoutRef = setTimeout(resolve, duration);
-    // don't keep process open
-    if (timeoutRef.unref && unref) {
+  private async _updateAndQueue() {
```

Comment:
I'm having a bit of trouble understanding exactly what this method does. Could we add some TSDoc or line comments within the code to clarify its purpose and functionality?

Reply:
Added some documentation.